From d52a4c2018fa9478704a5a769fdcefc428b3a8fc Mon Sep 17 00:00:00 2001
From: Rodrigo Duarte Sousa <rduartes@redhat.com>
Date: Tue, 18 Apr 2017 11:05:43 -0300
Subject: [PATCH] Unsetting the security_compliance flag

Turns out we need additional changes in "keystone.conf" file to make
it work properly and this should not be enabled by default.

Closes-bug: bz#1443133

Change-Id: Ibf0ea806f5c76a14372fa0c08cdef308245a6cbe
(cherry picked from commit baed4411fe8fa78959bcd9a5516a2ee5d447dfd6)
---
 config_tempest/config_tempest.py            | 13 +++++++++++++
 config_tempest/tests/test_config_tempest.py |  2 +-
 2 files changed, 14 insertions(+), 1 deletion(-)

diff --git a/config_tempest/config_tempest.py b/config_tempest/config_tempest.py
index 79ffc7d..2e560d1 100755
--- a/config_tempest/config_tempest.py
+++ b/config_tempest/config_tempest.py
@@ -724,6 +724,19 @@ def create_tempest_networks(clients, conf, has_neutron, public_network_id):
                         ' must be specified')
 
 
+def configure_keystone_feature_flags(conf, services):
+    """Set keystone feature flags based upon version ID."""
+    supported_versions = services.get('identity', {}).get('versions', [])
+    for version in supported_versions:
+        major, minor = version.split('.')[:2]
+        # Enable the domain specific roles feature flag. For more information,
+        # see https://developer.openstack.org/api-ref/identity/v3
+        if major == 'v3' and int(minor) >= 6:
+            conf.set('identity-feature-enabled',
+                     'forbid_global_implied_dsr',
+                     'True')
+
+
 def configure_boto(conf, services):
     """Set boto URLs based on discovered APIs."""
     if 'ec2' in services:
diff --git a/config_tempest/tests/test_config_tempest.py b/config_tempest/tests/test_config_tempest.py
index 68d6894..cc93c57 100644
--- a/config_tempest/tests/test_config_tempest.py
+++ b/config_tempest/tests/test_config_tempest.py
@@ -1,3 +1,4 @@
+
 # -*- coding: utf-8 -*-
 
 # Licensed under the Apache License, Version 2.0 (the "License"); you may
@@ -15,7 +16,6 @@
 """
 test_config_tempest
 ----------------------------------
-
 Tests for `config_tempest` module.
 """
 
